src/main/kotlin/no/skatteetaten/aurora/databasehotel/web/security/WebSecurityConfig.kt

```diff
@@ -0,0 +1,61 @@
+package no.skatteetaten.aurora.databasehotel.web.security
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationProvider
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken
+import org.springframework.security.web.authentication.preauth.RequestHeaderAuthenticationFilter
+import org.springframework.security.web.util.matcher.RequestMatcher
+
+@EnableWebSecurity
+@Configuration
+class WebSecurityConfig(
+    val sharedSecretReader: SharedSecretReader,
+    @Value("\${management.server.port}") val managementPort: Int,
+    @Value("\${aurora.authentication.enabled:true}") val authenticationEnabled: Boolean,
+    @Value("\${aurora.authentication.token.header-value-prefix:bearer|aurora-token}") val headerValuePrefixesRegexFormat: String
```

Comment:
Jeg må innrømme at jeg ikke klare lese dette uten å teste;

dbh i dag tar
Authorization: Bearer aurora-token aløsdjfaølsjdfa
men skal nå også støtte
Authorization: Bearer aløsdjfaølsjdfa
samtidig

Reply:
Slik jeg leser koden til den gamle authenticationManageren og application.yaml så har ikke dbh krevd Bearer prefix, kun aurora-token. 
I oppgaven står det spesifisert at dbh skal støtte både aurora-token og bearer som prefix, altså være bakoverkompatibel fram til vi får oppdatert konsumentene.
Har du et annet forslag?

Reply:
Kan også ta å legge inn en todo, evt en oppgave eller legge inn i oppgaven at det skal fjernes slik at bearer er det eneste som støttes.

Reply:
hehe... poenget mitt er bare det at dette må fungere med eksisterende boober/gobo, samtidig som man legger til rette for nytt format. Jeg klarer ikke lese ut fra koden om det er tilfelle, så hvis du ikke er 100% sikker, så må det testes.

Hvis du ser her: https://github.com/Skatteetaten/boober/blob/master/src/main/kotlin/no/skatteetaten/aurora/boober/service/resourceprovisioning/DatabaseSchemaProvisioner.kt#L113

så brukes Aurora-resttemplaten. Den er konfigurert her:
https://github.com/Skatteetaten/boober/blob/master/src/main/kotlin/no/skatteetaten/aurora/boober/Configuration.kt#L209

Som du ser så setter den Authorization: Bearer aurora-token <token> i dag. Og dette fungerer. Du må bare være sikker på at dette vil fortsette å fungere når du er ferdig - uavhengig av hva vi mener koden gjør ved å lese den.

Reply:
Da skjønner jeg, tydelig at det er mangel på tester fra min side. Skal ta å få lagd de slik at vi verifiserer at det er bakoverkompatibelt.

Reply:
:clap: